src/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs

```diff
@@ -0,0 +1,176 @@
+    {
+        private CommandOption _versionOption;
+        private CommandOption _frameworkOption;
+        private CommandOption _noRestoreOption;
+        private CommandOption _sourceOption;
+        private CommandOption _packageDirectoryOption;
+
+        public static DotNetSubCommandBase Create()
+        {
+            var command = new AddPackageReferenceCommand
+            {
+                Name = "package",
+                FullName = LocalizableStrings.AppFullName,
+                Description = LocalizableStrings.AppDescription,
+                HandleRemainingArguments = true,
+                ArgumentSeparatorHelpText = LocalizableStrings.AppHelpText,
+            };
+
+            command.HelpOption("-h|--help");
+
+            command._versionOption = command.Option(
+                $"-v|--version <{LocalizableStrings.CmdVersion}>",
+                LocalizableStrings.CmdVersionDescription,
+                CommandOptionType.SingleValue);
+
+            command._frameworkOption = command.Option(
+               $"-f|--framework <{LocalizableStrings.CmdFramework}>",
+               LocalizableStrings.CmdFrameworkDescription,
+               CommandOptionType.SingleValue);
+
+            command._noRestoreOption = command.Option(
+                "-n|--no-restore ",
+               LocalizableStrings.CmdNoRestoreDescription,
+               CommandOptionType.NoValue);
+
+            command._sourceOption = command.Option(
+                $"-s|--source <{LocalizableStrings.CmdSource}>",
+                LocalizableStrings.CmdSourceDescription,
+                CommandOptionType.SingleValue);
+
+            command._packageDirectoryOption = command.Option(
+                $"--package-directory <{LocalizableStrings.CmdPackageDirectory}>",
+                LocalizableStrings.CmdPackageDirectoryDescription,
+                CommandOptionType.SingleValue);
+
+            return command;
+        }
+
+        public override int Run(string fileOrDirectory)
+        {
+            if (RemainingArguments.Count != 1)
+            {
+                throw new GracefulException(LocalizableStrings.SpecifyExactlyOnePackageReference);
+            }
+
+            var projectFilePath = string.Empty;
+
+            if (!File.Exists(fileOrDirectory))
+            {
+                projectFilePath = MsbuildProject.GetProjectFileFromDirectory(fileOrDirectory).FullName;
+            }
+            else
+            {
+                projectFilePath = fileOrDirectory;
+            }
+
+            var tempDgFilePath = string.Empty;
+
+            if (!_noRestoreOption.HasValue())
+            {
+                // Create a Dependency Graph file for the project
+                tempDgFilePath = Path.GetTempFileName();
+                GetProjectDependencyGraph(projectFilePath, tempDgFilePath);
+            }
+
+            var result = NuGetCommand.Run(TransformArgs(RemainingArguments.First(), tempDgFilePath, projectFilePath));
```

Comment:
I would assign RemainingArguments.First() to a packageToAdd variable, just to make the code clearer.